lib/gui/components/drive-selector/templates/drive-selector-modal.tpl.html

```diff
@@ -15,35 +15,15 @@ <h4 class="list-group-item-heading">{{ drive.description }} - {{ drive.size | gi
 
           <footer class="list-group-item-footer">
 
-            <span class="label label-warning"
-                ng-show="modal.constraints.isDriveLargeEnough(drive, modal.state.getImage()) && !modal.constraints.isDriveLocked(drive) && !modal.constraints.isDriveSizeRecommended(drive, modal.state.getImage())">
-                <i class="glyphicon glyphicon-warning-sign"></i>
-                Not Recommended</span>
-
-            <!-- There can be a case where the device it not large enough, and it's also locked. -->
-            <!-- Since in this case both labels will be displayed, we chose to only show the -->
-            <!-- "not large enough label", since from the point of view of the user, the locked -->
-            <!-- state is irrelevant if the drive is not large enough anyway. -->
-
-            <span class="label label-danger"
-              ng-hide="modal.constraints.isDriveLargeEnough(drive, modal.state.getImage())">
-                <i class="glyphicon glyphicon-resize-small"></i>
-                Too Small For Image</span>
-
-            <span class="label label-danger"
-              ng-show="modal.constraints.isDriveLocked(drive) && modal.constraints.isDriveLargeEnough(drive, modal.state.getImage())">
-                <i class="glyphicon glyphicon-lock"></i>
-                Locked</span>
-
-            <span class="label label-danger"
-              ng-show="modal.constraints.isSystemDrive(drive)">
-                <i class="glyphicon glyphicon-hdd"></i>
-                System Drive</span>
-
-            <span class="label label-danger"
-              ng-show="modal.constraints.isSourceDrive(drive, modal.state.getImage())">
-                <i class="glyphicon glyphicon-circle-arrow-down"></i>
-                Source Drive</span>
+            <span class="label"
+              ng-class="{
+                'label-warning': modal.getDriveImageCompatibilityStatus(drive).type === modal.constraints.COMPATIBILITY_STATUS_TYPES.WARNING,
+                'label-warning': modal.constraints.isDriveValid(drive, modal.state.getImage()) && modal.getDriveImageCompatibilityStatus(drive).message === modal.constraints.COMPATIBILITY_STATUS_MESSAGES.SYSTEM,
```

Comment:
@Shou I think this line is unnecessary if the fact that a drive is a system drive is made a warning rather than an error in the drive constraints file, which sounds logical IMHO.